pkg/propertyprovider/aks/provider.go

```diff
@@ -119,25 +118,32 @@ func (p *PropertyProvider) Start(ctx context.Context, config *rest.Config) error
 		return err
 	}
 
-	if p.region == nil {
-		klog.V(2).Info("Auto-discover region as none has been specified")
-		// Note that an API reader is passed here for the purpose of auto-discovering region
-		// information from AKS nodes; at this time the cache from the controller manager
-		// has not been initialized yet and as a result cached client is not yet available.
-		//
-		// This incurs the slightly higher overhead, however, as auto-discovery runs only
-		// once, the performance impact is negligible.
-		discoveredRegion, err := p.autoDiscoverRegionAndSetupTrackers(ctx, mgr.GetAPIReader())
-		if err != nil {
-			klog.ErrorS(err, "Failed to auto-discover region for the AKS property provider")
-			return err
+	klog.V(2).Info("Setting up the node tracker")
+	if p.nodeTracker == nil {
```

Comment:
when will the node Tracker be not nil?

Reply:
Ah, this is for the case where a pricing provider is plugged in directly. Node tracker will be created here in the New function if no pricing provider is provided.